README.md

```diff
@@ -1,199 +1,145 @@
-
-- data type: cmip26.py
-- training type: trainScript.py
-- testing type: testing/main.py
-
-The data type experiment corresponds to different runs of the scirpt cmip26.py generating the coarse surface velocities, as well as the diagnosed forcing.
-
-The training type experiment corresponds to different runs of trainScript.py, resulting in a trained model. Various runs might used different neural networks, loss
-functions, number of training epochs, or training data. For the latter, it makes sense that one will need to specify a run from the data type experiment
-where the training data will be collected for training. Different runs are also used for different values of CO2.
-
-The testing type uses a model from the training type experiment and applies it, without further training, to global data. The global data used for the test
-is, again, specified as a run from the data type experiment.
+_This repository is currently work-in-progress following a process of
+refreshing the code and making it available for easy reuse by others._
 
-## Specific to Greene implementation
+## Architecture
+The model is written in Python, using PyTorch for the CNN. We provide 3 separate
+"stages", which are run using different commands and arguments:
 
-To list the currently existing experiments, one can follow this procedure:
-
-- Start a singularity container with the appropriate image via the following command:
-```
-singularity exec --nv --overlay /scratch/ag7531/overlay2 /scratch/work/public/singularity/cuda11.0-cudnn8-devel-ubuntu18.04.sif /bin/bash
-```
-- Activate conda:
-```
-source /ext3/env.sh
-```
-- Activate the subgrid conda environment:
-```
-conda activate subgrid
-```
+* data processing: downloads part of CM2.6 dataset and processes
+* model training: train model on processed data
+* model testing: tests the trained model on an unseen region
 
-Finally, start python, import the mlflow module and call the mlflow method that allows to list expriments:
-```
-from mlflow.tracking import client
-cl = client.MlflowClient()
-for e in cl.list_experiments():
-    print(e.experiment_id, e.name)
-```
+For more details on each of the stages, see the `docs` directory.
 
-If nothing shows, return to bash, create the following environment variable to tell mlflow where to look for experiments,
-```
-export MLFLOW_TRACKING_URI=/scratch/ag7531/mlruns
-```
+## Usage
+### Dependencies
+Python 3 is required.
 
-and run the python lines again.
+#### Python
+With `pip` installed, run the following in the root directory:
 
-Right now, there are many experiments, including some old ones which are no more used. But the most important ones are:
-- for data: data-global, with id 19
-- for models: modelsv1, with id 21
-- for testing: test_global, with id 20
+    pip install -e
 
-# Run data code
+*(An alternate `pyproject.toml` file is provided for building with Poetry. To
+use, rename `pyproject-poetry.toml` to `pyproject.toml` (clobbering the existing
```

Comment:
Thanks, nice catch. It means to overwrite an existing file with a different one, usually by renaming. \*nix-only because you can't clobber in Windows (you get told "filename already exists). I think this is filesystem speak. I'll change to "overwrite" or so.